keyboards/cipulot/ec_23u/keymaps/via/via_apc.c

```diff
@@ -0,0 +1,156 @@
+/* Copyright 2023 Cipulot
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "ec_switch_matrix.h"
+#include "action.h"
+#include "via.h"
+
+void apc_init_thresholds(void);
+void apc_set_threshold(bool is_for_actuation);
+
+// Declaring an _apc_config_t struct that will store our data
+typedef struct _apc_config_t {
+    uint16_t actuation_threshold;
+    uint16_t release_threshold;
+} apc_config;
+
+// Check if the size of the reserved persistent memory is the same as the size of struct apc_config
+_Static_assert(sizeof(apc_config) == EECONFIG_KB_DATA_SIZE, "Mismatch in keyboard EECONFIG stored data");
+
+// Declaring a new variable apc of type apc_config
+apc_config apc;
+
+// Declaring enums for VIA config menu
+enum via_apc_enums {
+    // clang-format off
+    id_apc_actuation_threshold = 1,
+    id_apc_release_threshold = 2
+    // clang-format on
+};
+
+// Initializing persistent memory configuration: default values are declared and stored in PMEM
+void eeconfig_init_kb(void) {
+    // Default values
+    apc.actuation_threshold = DEFAULT_ACTUATION_LEVEL;
+    apc.release_threshold   = DEFAULT_RELEASE_LEVEL;
+    // Write default value to EEPROM now
+    eeconfig_update_kb_datablock(&apc);
```

Comment:
Nor should it be using the kb eeprom block

Reply:
As before.